### PR TITLE
meta quest full body tracking

### PR DESCRIPTION
**Pull request by barnabwhy**
to-do:
- [x] get body tracking data from headset to alvr
- [x] send data to osc
- [x] emulate vive trackers
- [x] properly handle recentering for osc data ~~(no clue how to do this)~~
- [x] properly handle recentering for fake vive trackers
- [x] support the undocumented `XR_META_body_tracking_full_body` extension (see https://github.com/alvr-org/openxrs/pull/3)

### CODE REVIEW

**Comment by zarik5** [comment]
Just so you are aware, this is going to be a breaking change. If you want to implement this without breaking the protocol, you can emulate vive trackers on the client side and send them through the `Tracking::device_motions` vector.

**Comment by barnabwhy** [comment]
> Just so you are aware, this is going to be a breaking change. If you want to implement this without breaking the protocol, you can emulate vive trackers on the client side and send them through the `Tracking::device_motions` vector. You can use the OpenXR paths defined in `XR_HTCX_vive_tracker_interaction`.

i've (hopefully this is okay) opted to instead just use some generic paths since it isn't specifically for vive trackers and more generally body tracking. i guess eventually when body tracking becomes standard in openxr it'll use some generic paths anyway?

**Comment by barnabwhy** [comment]
needs alvr-org/openxrs#3

File: alvr/server/cpp/alvr_server/FakeViveTracker.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#include "FakeViveTracker.h"
+#include "Settings.h"
+
+#include <cassert>
+#include <format>
```
Comment (wsippel) [inline alvr/server/cpp/alvr_server/FakeViveTracker.cpp:5]: Those two includes appear to be unused? Also, `#include <format>` causes the Linux build to fail as far as I can tell.

**Review by zarik5 — COMMENTED** [review]
Overall looks good, except that I don't think you need to support both `XR_FB_body_tracking` and `XR_META_body_tracking_full_body` extensions. The META one will suffice.

**Comment by barnabwhy** [comment]
> Overall looks good, except that I don't think you need to support both `XR_FB_body_tracking` and `XR_META_body_tracking_full_body` extensions. The META one will suffice.

if I'm not mistaken, generative legs was added 1 update later than iobt, so it possible that someone's headset only supports one (i think). plus the meta one doesn't allow you to disable the legs, only the other one

File: alvr/server/cpp/alvr_server/FakeViveTracker.cpp (hunk 1 above)
Comment (barnabwhy) [reply alvr/server/cpp/alvr_server/FakeViveTracker.cpp:5]: oh shit, yeah I'm not even using format i think vscode added that while i wasn't paying attention

**Comment by zarik5** [comment]
> so it possible that someone's headset only supports one (i think)

I think if The headset is updated it should support legs anyway. it's not based on sensors, it's just estimated, so even the Quest 2 should support it.

**Pull request merged**